### PR TITLE
Auto ci ms2snoop

### PR DESCRIPTION
**Pull request by Lain-inrae**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

**updates the PR CI**
 
feat(ms2snoop:xml+iuc): makes the xml more iuc compliant
fix(ms2snoop:mainloop): fixes some errors in ms2snoop mainloop
fix(ms2snoop:dependencies): fix optparse version

### CODE REVIEW

**Comment by Lain-inrae** [comment]
@fgiacomoni

**Comment by Lain-inrae** [comment]
Screenshot 2022-05-05 at 16-10-10 https __pipelines actions githubusercontent com
Screenshot 2022-05-05 at 16-59-41 https __pipelines actions githubusercontent com

**Comment by Lain-inrae** [comment]
The CI says 
```yml
name: Galaxy Tool Linting and Tests for push and PR
on: [push, pull_request]
env:
  GALAXY_REPO: https://github.com/galaxyproject/galaxy
  GALAXY_RELEASE: release_20.09
  MAX_CHUNKS: 4
  ```
  I suppose release_20.09 shoud be upgraded to 22.01 or something like that.

**Comment by fgiacomoni** [comment]
@Lain-inrae thank you for the fix and updates linked with previous review made by @lecorguille 
PR validated from my side for the tool and its wrapper. 
Can't really help on the CI error... Any idea from the Galaxy folks ?

**Comment by Lain-inrae** [comment]
gxformat2

**Comment by Lain-inrae** [comment]
In The CI:

gxformat2-ci
planemo-ci

can't explain....

**Comment by Lain-inrae** [comment]
gxformat2-ci

**Comment by Lain-inrae** [comment]
@lecorguille 
The CI still crashes even without the profile setting in the xml file.
Do you have any idea?

**Comment by fgiacomoni** [comment]
This thread was opened 1 year ago: https://githubhot.com/repo/galaxyproject/planemo/issues/1147 May be can help

**Comment by Lain-inrae** [comment]
Yup, the issue is about planemo 0.74.3. Here, we have planemo 0.74.9. So, not sure if if applies... But its clearly related.

**Comment by bgruening** [comment]
I recommend updating the CI integration: https://github.com/workflow4metabolomics/tools-metabolomics/blob/master/.github/workflows/pr.yaml

Its over a year old. Take the latest one from IUC and we can have a look if the error still exists.

**Comment by Lain-inrae** [comment]
Yaaaaaayyy!!! 
With the pr.yaml file updated everything work!
That's great, thanks Björn!

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -368,70 +326,79 @@ jobs:
...
+      uses: galaxyproject/planemo-ci-action@v1
+      id: check
+      with:
+        mode: check
 
   # deploy the tools to the toolsheds (first TTS for testing)
   deploy:
     name: Deploy
     needs: [setup, lint, flake8, lintr, combine_outputs]
+    if: ${{ (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/main' ) && github.repository_owner == 'galaxyproject' }}
```
Comment (bgruening) [inline .github/workflows/pr.yaml:350]: galaxyproject needs to be adapted to `workflow4metabolomics`

File: tools/ms2snoop/MS2snoop.xml [hunk 2]
```diff
@@ -1,12 +1,37 @@
-<tool id="ms2snoop" name="MS2 Snoop" version="1.0.0" profile="21.05">
+<tool id="ms2snoop" name="MS2 Snoop" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="22.01">
+
+    <macros>
+        <token name="@TOOL_VERSION@">1.0.1</token>
+        <token name="@VERSION_SUFFIX@">1</token>
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:5]: Suffix can be 0 since this is your first version of 1.0.1

File: tools/ms2snoop/MS2snoop.xml [hunk 3]
```diff
@@ -1,12 +1,37 @@
...
+        <requirement type="package" version="1.7.1">r-optparse</requirement>
     </requirements>
 
+    <stdio>
+        <exit_code range="1:" />
+        <regex match="Error in\s+.*:\s+.*" />
+    </stdio>
+
+    <version_command>
+        Rscript $__tool_directory__/MS2snoop.R --version | head -n 1
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:32]: Single quotes are missing

**Comment by Lain-inrae** [comment]
I hope the quality of the tool/wrapper has reached a good level.
If it's not the case, please tell me! (I'd like to have an example tool/wrapper to reuse for some colleagues)

File: tools/ms2snoop/MS2snoop.xml [hunk 4]
```diff
@@ -1,117 +1,215 @@
...
+    </macros>
+
     <description>
         MS1/MS2 spectra and associated adducts extraction and cleaning
     </description>
 
+    <edam_topics>
+        <edam_topic>topic_0091</edam_topic>
+        <edam_topic>topic_3370</edam_topic>
+    </edam_topics>
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:15]: I would remove those empty lines.

File: tools/ms2snoop/MS2snoop.xml [hunk 5]
```diff
@@ -1,117 +1,215 @@
...
-            min="0"
-            max="25"
-            value="8"
-            label="Present in at least X scan"
-            help="fragments are kept if they are found in a minimum number of scans"
-        />
+
+        <conditional name="param">
+
+            <param
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:79]: I think this parameter and this function is not needed for Galaxy. You see the version as part of the tool and if you run it also as part of the provenance.

File: tools/ms2snoop/MS2snoop.xml [hunk 6]
```diff
@@ -1,117 +1,215 @@
...
+                type="boolean"
+                value="false"
+                label="Prints version"
+                argument="--version"
+                help="The tool prints its version and then exits"
+            />
+
+            <when value="false">
+
+                <param
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:89]: on IUC we prefer the one-line syntax ... mostly

File: tools/ms2snoop/MS2snoop.xml [hunk 7]
```diff
@@ -1,117 +1,215 @@
...
+                    type="data"
+                    format="tabular"
+                    name="compound_txt"
+                    label="compounds_pos.txt"
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:93]: the label should be a description and not a file name

File: tools/ms2snoop/MS2snoop.xml [hunk 8]
```diff
@@ -1,117 +1,215 @@
...
+                />
+                <param
+                    type="data"
+                    format="tabular"
+                    name="peaklist_frag_tsv"
+                    label="peaklist_fragments.tsv"
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:99]: as part of the labels/help I would describe what the input should look like, how was it produced etc ...

File: tools/ms2snoop/MS2snoop.xml [hunk 9]
```diff
@@ -1,117 +1,215 @@
...
+                />
+                <param
+                    type="data"
+                    format="tabular"
+                    name="peaklist_preco_tsv"
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:104]: why not using `argument="--precursors"` instead ...

File: tools/ms2snoop/MS2snoop.xml [hunk 10]
```diff
@@ -1,117 +1,215 @@
...
+                <param
+                    type="integer"
+                    min="0"
+                    max="25"
+                    value="8"
+                    label="Present in at least X scan"
+                    argument="--min_number_scan"
+                    help="fragments are kept if they are found in a minimum number of scans"
+                />
+                <param
```
Comment (bgruening) [inline tools/ms2snoop/MS2snoop.xml:155]: verbose and debug I would not provide as a user choice, or at least hidden into an advanced section

**Review by bgruening — COMMENTED** [review]
Some comments since you ask for best practices?

File: tools/ms2snoop/MS2snoop.xml (hunk 9 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml:104]: planemo says we can't use "argument" along with type="data"...
Perhaps I made an error somewhere. I'll check that.

File: tools/ms2snoop/MS2snoop.xml (hunk 5 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml]: version check deleted

File: tools/ms2snoop/MS2snoop.xml (hunk 6 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml]: done!

File: tools/ms2snoop/MS2snoop.xml (hunk 8 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml]: fixed!

File: tools/ms2snoop/MS2snoop.xml (hunk 10 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml:123]: fixed

File: tools/ms2snoop/MS2snoop.xml (hunk 7 above)
Comment (Lain-inrae) [reply tools/ms2snoop/MS2snoop.xml]: fixed!

**Comment by Lain-inrae** [comment]
Thanks a lot! ^^

**Comment by Lain-inrae** [comment]
@bgruening sorry to bother - is this tool ready to be merged?

**Review by fgiacomoni — APPROVED** [review]
@Lain-inrae , thank you for all integrated comment
@bgruening,  megathanks for this review - really helpful for the whole W4M core team to build our future w4m tool template
Concerning EDAM selected concepts, not really agree with choosen operations: operation_3803 (natural product id) and operation_3860 (spectrum calculation)... Need to discuss with edam team to imporve the metabolomics part of the ontology.

**Pull request merged**